src/content/docs/identityserver/ui/login/dynamicproviders.md

```diff
@@ -156,37 +272,169 @@ class CustomOidcConfigureOptions : ConfigureAuthenticationOptions<OpenIdConnectO
+[shared configuration options](../../../reference/options#dynamic-providers) for the dynamic identity providers
+feature. For example, you can customize the path prefix for the dynamic providers callback path:
 
-As part of the architecture of the dynamic providers feature, the various callback paths are required and are
-automatically set to follow a convention.
-The convention of these paths follows the form of `~/federation/{scheme}/{suffix}`.
+```csharp title="Program.cs"
+builder.Services
+    .AddIdentityServer(options =>
+    {
+        // ...
+        
+        options.DynamicProviders.PathPrefix = "/fed";
+        
+        // ...
+    })
+```
 
-These are three paths that are set on the `OpenIdConnectOptions`:
+## Using Non-OIDC Authentication Handlers 
 
-* [CallbackPath](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.remoteauthenticationoptions.callbackpath).
-  This is the OIDC redirect URI protocol value. The suffix "/signin" is used for this path.
-* [SignedOutCallbackPath](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.openidconnect.openidconnectoptions.signedoutcallbackpath).
-  This is the OIDC post logout redirect URI protocol value. The suffix "/signout-callback" is used for this path.
-* [RemoteSignOutPath](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.authentication.openidconnect.openidconnectoptions.remotesignoutpath).
-  This is the OIDC front channel logout URI protocol value. The suffix "/signout" is used for this path.
+Dynamic identity providers in Duende IdentityServer come with an implementation that supports OpenId Connect providers to be registered.
+In your solution, it may be necessary to support other authentication providers, such as the `GoogleHandler`, or a SAML-based authentication provider.
 
-This means for your IdentityServer running at "https://sample.duendesoftware.com" and an OIDC identity provider whose
-scheme is "idp1", your client configuration with the external OIDC identity provider would be:
+To register other authentication handlers, you can use the `AddProviderType<T, TOptions, TIdentityProvider>(string scheme)` method on the `DynamicProviderOptions` object,
+where `T` is the authentication handler type, `TOptions` is the options type for that particular handler, and `TIdentityProvider` is the identity provider type that models the dynamic provider.
 
-* The redirect URI would be "https://sample.duendesoftware.com/federation/idp1/signin"
-* The post logout redirect URI would be "https://sample.duendesoftware.com/federation/idp1/signout-callback"
-* The front channel logout URI would be "https://sample.duendesoftware.com/federation/idp1/signout"
+The authentication handler type and options type will typically be provided by the authentication provider itself.
+For example, the `GoogleHandler` and `GoogleOptions` types are provided by the `Google.AspNetCore.Authentication.OAuth` NuGet package.
```

Comment:
When would anyone actually use multiple Google handlers?

Would it make sense to use a Saml2 example instead? https://docs.duendesoftware.com/identityserver/samples/ui/#using-sustainsyssaml2-with-dynamic-providers

Reply:
Multi-tenant scenarios where the tenant is responsible for configuring their IdPs (at least this is one where I have used them). Added a link to the Saml2 example, that's a good one.

Reply:
Oh, I tend to give Google and Entra ID special treatment - a single scheme each for those that's shared between all tenants. And then tenant-specific OIDC + Saml2 schemes for anyone having on-prem/custom provider.

Reply:
You OK with the mention as it is right now?